File: LC/python/Merge Intervals.py
# Definition for an interval.
# class Interval:
#     def __init__(self, s=0, e=0):
#         self.start = s
#         self.end = e

class Solution:
    # @param intervals, a list of Interval
    # @return a list of Interval
    def merge(self, intervals):
        intervals.sort(key = lambda x:x.start)
        res = []
        for i in range(len(intervals)):
            if res == []:
                res.append(intervals[i])
            else:
                if res[len(res)-1].start<= intervals[i].start <= res[len(res)-1].end:
                    res[len(res)-1].end = max(res[len(res)-1].end,intervals[i].end)
                else:
                    res.append(intervals[i])
        return res

File: LC/java/Merge Intervals.java
/**
 * Definition for an interval.
 * public class Interval {
 *     int start;
 *     int end;
 *     Interval() { start = 0; end = 0; }
 *     Interval(int s, int e) { start = s; end = e; }
 * }
 */
public class Solution {
    public ArrayList<Interval> merge(ArrayList<Interval> intervals) {
        ArrayList<Interval> res = new ArrayList<Interval>();
        
        Collections.sort(intervals, new Comparator<Interval>(){
            public int compare(Interval a,Interval b){
                return a.start>b.start?1:(a.start==b.start?0:-1);}});
        
        if (intervals == null || intervals.size()<2){
            return intervals;
        }
        
        for(Interval inter : intervals){
            if (res.size()==0){
                res.add(inter);
            }
            else{
                if(res.get(res.size()-1).start<=inter.start && inter.start<=res.get(res.size()-1).end){
                    res.get(res.size()-1).end = Math.max(res.get(res.size()-1).end,inter.end);
                }
                else{
                    res.add(inter);
                }
            }
        }
        return res;
    }
}